handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -255,10 +249,20 @@ protected void deallocate() {
             if (enableOcsp) {
                 SSL.enableOcsp(ssl);
             }
+
+            if (!jdkCompatibilityMode) {
+                SSL.setMode(ssl, SSL.getMode(ssl) | SSL.SSL_MODE_ENABLE_PARTIAL_WRITE);
+            }
+
+            // setMode may impact the overhead.
+            maxWrapOverhead = SSL.getMaxWrapOverhead(ssl);
         } catch (Throwable cause) {
             SSL.freeSSL(ssl);
             PlatformDependent.throwException(cause);
         }
+        this.maxWrapOverhead = maxWrapOverhead;
+        // maxWrapBufferSize must be set after maxWrapOverhead because there maybe a dependency on this value.
+        maxWrapBufferSize = jdkCompatibilityMode ? maxEncryptedPacketLength() : maxEncryptedPacketLength() << 4;
```

Comment:
done ... a bit of a magic number ... the comment says we can make this configurable in the future if necessary. for now I think its fine to start with enough space for 16 TLS packets per wrap.